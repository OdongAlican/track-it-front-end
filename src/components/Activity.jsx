/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable jsx-a11y/no-static-element-interactions */

import React from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { deleteActivity } from '../actions/activitiesAction';

const Activity = ({ activity }) => {
  const dispatch = useDispatch();

  const deleteAct = id => {
    dispatch(deleteActivity(id));
  };

  return (
    <div data-testid="appActivity" className="main-section d-flex p-1 bg-white">
      <div className="image-section">
        <img src={activity.avatar.url} alt="boohoo" className="image-det" />
      </div>
      <div className="activity-body ml-1">
        <div className="w-100 title-div-section">
          <div className="activity-title text-secondary d-flex">
            { activity.title.length < 15 ? `${activity.title}`
              : `${activity.title.substring(0, 20)}...` }
          </div>
        </div>
        <div className="activity-text text-secondary d-flex">
          <div className="activity-inner">
            { activity.total }
          </div>
          <div>
            <small>Hrs</small>
          </div>
        </div>
        <div className="lower-section">
          <Link
            className="links-info"
            to={{
              pathname: `/activity/${activity.id}/measurements`,
              state: activity,
            }}
          >
            <button
              type="button"
              className="button-details"
            >
              Details
            </button>
          </Link>
          <div className="activity-link">
            <i className="far fa-trash-alt trash-section text-danger" onClick={() => deleteAct(activity.id)} />
          </div>
        </div>
      </div>
    </div>
  );
};

Activity.propTypes = {
  activity: PropTypes.shape({
    id: PropTypes.number,
    avatar: PropTypes.objectOf(PropTypes.string),
    title: PropTypes.string,
    total: PropTypes.string,
  }).isRequired,
};

export default Activity;
