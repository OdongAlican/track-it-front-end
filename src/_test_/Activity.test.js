import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Activity from '../components/Activity';

afterEach(cleanup);
const initialState = {
  authReducer: {
    output: 10,
    error: 'test',
  },
};

const mockStore = configureStore();
let store;

const activity = {
  title: 'test title',
  total: 'test total',
  avatar: {
    url: 'sample',
  },
};

it('renders Activity correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Activity activity={activity} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appActivity');
  expect(result[0]).toHaveTextContent('test titletest totalHrsDetails');
});

it('renders Activity correctly', () => {
  store = mockStore(initialState);
  const { getAllByTestId } = render(
    <Router>
      <Provider store={store}><Activity activity={activity} /></Provider>
      ,
    </Router>,
  );
  const result = getAllByTestId('appActivity');
  expect(result[0]).not.toHaveTextContent('some random text');
});

it('matches Activity snapshot', () => {
  const appointment = renderer.create(
    <Router>
      <Provider store={store}><Activity activity={activity} /></Provider>
      ,
    </Router>,
  );
  const tree = appointment.toJSON();
  expect(tree).toMatchSnapshot();
});
