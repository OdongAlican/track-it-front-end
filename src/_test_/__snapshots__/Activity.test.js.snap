// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches Activity snapshot 1`] = `
Array [
  <div
    className="main-section d-flex p-1 bg-white"
    data-testid="appActivity"
  >
    <div
      className="image-section"
    >
      <img
        alt="boohoo"
        className="image-det"
        src="sample"
      />
    </div>
    <div
      className="activity-body ml-1"
    >
      <div
        className="w-100 title-div-section"
      >
        <div
          className="activity-title text-secondary d-flex"
        >
          test title
        </div>
      </div>
      <div
        className="activity-text text-secondary d-flex"
      >
        <div
          className="activity-inner"
        >
          test total
        </div>
        <div>
          <small>
            Hrs
          </small>
        </div>
      </div>
      <div
        className="lower-section"
      >
        <a
          className="links-info"
          href="/activity/undefined/measurements"
          onClick={[Function]}
        >
          <button
            className="button-details"
            type="button"
          >
            Details
          </button>
        </a>
        <div
          className="activity-link"
        >
          <i
            className="far fa-trash-alt trash-section text-danger"
            onClick={[Function]}
          />
        </div>
      </div>
    </div>
  </div>,
  ",",
]
`;
